contributed/matmul.py

```diff
@@ -0,0 +1,131 @@
+from neuronxcc import nki
+import neuronxcc.nki.language as nl
+import neuronxcc.nki.isa as ni
+import numpy as np
+
+def matmul(A_DRAM, B_DRAM, Z_DRAM):
+  """
+  Optimized matrix multiplication kernel
+
+   Args:
+
+      A_DRAM: an input tensor of shape [K, M], where K is a multiple of 1024
+      and M is a multiple of 512.  It is the left-hand-side argument of the
+      matrix multiplication, delivered transposed for optimal performance.
+
+      B_DRAM: an input tensor of shape [K, N],  where K is a multiple of 1024
+        and N is a multiple of 2048.  It is the right-hand-side argument of
+        the matrix multiplication.
+
+      result: the resulting output tensor of shape [M, N]
+
+  """
+  K, M = A_DRAM.shape
+  _, N = B_DRAM.shape
+
+  TILE_K = nl.tile_size.pmax
+  TILE_M = nl.tile_size.gemm_stationary_fmax
+  TILE_N = nl.tile_size.gemm_moving_fmax
+
+  TILES_IN_BLOCK_M = 4
```

Comment:
These meta parameters can be in the kernel arguments. It helps with handling inputs with other sizes later.

Reply:
This should be fixed. Thanks!